lib/buffer.js

```diff
@@ -42,34 +42,92 @@ function alignPool() {
   }
 }
 
-
-function Buffer(arg, encoding) {
+/**
+ * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
+ * in the documentation and should not be used moving forward. Rather,
+ * developers should use one of the three new factory APIs: Buffer.from(),
+ * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
+ * is no hard deprecation because of the extent to which the Buffer constructor
+ * is used in the ecosystem currently -- a hard deprecation would introduce too
+ * much breakage at this time. Eventually, however, we should re-evaluate that
+ * choice.
```

Comment:
Well, if we made the change we'd have to do parallel releases in each of the downlevel streams (v4, v0.12 and v0.10) that also zero filled by default. You're right, let's hold off on that one a bit longer.